src/components/PopupWithForm.js

```diff
@@ -22,6 +23,14 @@ export default class PopupWithForm extends Popup {
     return inputsItems;
   }
 
+  buttonLoading(loading) {
+    if (loading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
```

Comment:
Needs correction: here you should return the initial button text. For that you can save it in the constructor and use here:

```js
// in the constructor
this._buttonText = this._submitButton.textContent;
```